File: {{ cookiecutter.repo_name }}/{{ cookiecutter.module_name }}/analyze/analysis.py
from pathlib import Path

from loguru import logger
from tqdm import tqdm

from {{ cookiecutter.module_name }}.config import PROCESSED_DATA_DIR


def main(
    input_path: Path = PROCESSED_DATA_DIR / "dataset.csv",
    output_path: Path = PROCESSED_DATA_DIR / "analysis_results.csv",
) -> None:
    """Perform analysis on processed dataset.

    Parameters
    ----------
    input_path : Path
        Path to input dataset.
    output_path : Path
        Path to save analysis results.
    """
    logger.info("Analyzing dataset...")
    for i in tqdm(range(10), total=10):
        if i == 5:
            logger.info("Something happened for iteration 5.")
    logger.success("Analysis complete.")


if __name__ == "__main__":
    main()

File: {{ cookiecutter.repo_name }}/src/analyze/analysis.jl
"""
Perform analysis on processed dataset.
Mirrors {{ cookiecutter.module_name }}/analyze/analysis.py
"""

using Logging

include("../config.jl")

const INPUT_PATH = joinpath(PROCESSED_DATA_DIR, "dataset.arrow")
const OUTPUT_PATH = joinpath(PROCESSED_DATA_DIR, "analysis_results.arrow")

@info "Analyzing dataset..."

for i in 1:10
    if i == 5
        @info "Something happened for iteration 5."
    end
end

@info "Analysis complete."
